Slim/Plugin/RandomPlay/Mixer.pm

```diff
@@ -170,6 +170,8 @@ sub getRandomYear {
 		$years = [ Slim::Schema->rs('Track')->search(\%cond, \%attr)->get_column('me.year')->all ];
 	}
 
+	Slim::Player::Playlist::fischer_yates_shuffle($years); # SQLite's RAND function is not very random. This does a better job.
```

Comment:
The you could probably get rid of the sorgin in the query to save a few CPU cycles? Years lists usually aren't long, though 😀 .